src/server/req_quejob.c

```diff
@@ -271,27 +271,40 @@ get_server_index(void)
  * 			Note: Consider directly modifying get_next_svr_sequence_id() if reservations
  * 			also get sharded for multi-server
  *
- * @param[out]	jidbuf - buffer to fill job id in
+ * @param[out]	idbuf - buffer to fill job/resv id in
  * @param[in]	clusterid - cluster name (PBS_SERVER)
- * @param[in]	jtype - job type (0 for normal job, 1 for job array)
+ * @param[in]	objtype - object type (0 for normal job, 1 for job array, 2 for reservation)
+ * @param[in]	resv_char - character representing type of reservation
  *
  * @return	int
  * @retval	0 for Success
  * @retval	1 for Failure
  */
 static int
-generate_jobid(char *jidbuf, char *clusterid, int jtype)
+generate_objid(char *idbuf, char *clusterid, int objtype, char resv_char)
```

Comment:
to keep the naming consistent, please rename "get_job_resv_location_hint" to "get_obj_location_hint"

Reply:
I now named this function as get_shard_obj_location_hint(). This is very generic and since we need this function only for sharded objects like jobs/resvs, I have named it like this.

Reply:
then you should name this "generate_shard_objid" since the same logic applies here. TBH, I think it is clear enough even if you don't use the word 'shard'. And a simple if condition inside generate_obj_loation_hint() can ensure that it handles other objects as well (which will also help you remove similar if conditions before calling generate_obj_loation_hint())